src/core/attack_generation.h

```diff
@@ -111,35 +86,28 @@ constexpr static inline uint64_t getKingAttacks(const BitBoard& board)
     return movegen::getKingMoves(utils::lsbToPosition(king));
 }
 
-constexpr static inline uint64_t getWhitePawnAttacks(const BitBoard& board)
+template<Player player>
+constexpr static inline uint64_t getPawnAttacks(const BitBoard& board)
 {
-    const uint64_t pawns = board.pieces[WhitePawn];
-    uint64_t attacks = ((pawns & ~s_aFileMask) << 7);
-    attacks |= ((pawns & ~s_hFileMask) << 9);
-
-    return attacks;
-}
+    if constexpr (player == PlayerWhite) {
+        const uint64_t pawns = board.pieces[WhitePawn];
+        uint64_t attacks = ((pawns & ~s_aFileMask) << 7);
+        attacks |= ((pawns & ~s_hFileMask) << 9);
 
-constexpr static inline uint64_t getBlackPawnAttacks(const BitBoard& board)
-{
-    const uint64_t pawns = board.pieces[BlackPawn];
-    uint64_t attacks = ((pawns & ~s_aFileMask) >> 9);
-    attacks |= ((pawns & ~s_hFileMask) >> 7);
+        return attacks;
```

Comment:
Nitpick: formatting

Reply:
clang-formatted - just looks weird on Github due to moved into the scope :)